core/templates/dev/head/domain/collection/CollectionUpdateService.js

```diff
@@ -257,21 +279,30 @@ oppia.factory('CollectionUpdateService', [
         },
 
         /**
+         * Returns whether the collection tag is valid.
+         */
+        isTagValid: function(collection, tags) {
```

Comment:
done! my implementation does seem clunky, so I would like extra feedback on it. (the reason it is clunky is due to the way I had to link `alertService`.